From 5d1d1337305f5d2ee8772b34e8ebf56e43ad482a Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Mon, 27 Jan 2025 17:07:43 +0100
Subject: [PATCH 131/140] mailbox: stm32-ippc: change PM level for IPI

The SCMI resource, as clock, are based on IPCC mailbox and are use
d during suspend/resume phase of other device, so the IPPC used as IPI
must be resumed in the early stage.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/linux.git 790cc234b2a0ce7414337de38b92bb0f9a03d6ad mailbox: stm32-ippc: change PM level for IPI]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Ibd3cf3024c21d06bb99ac29b6fba8fe5cfa4c83c
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/432662
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/mailbox/stm32-ipcc.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/mailbox/stm32-ipcc.c b/drivers/mailbox/stm32-ipcc.c
index 2c8e93a9e6ebd..ef9e04f086193 100644
--- a/drivers/mailbox/stm32-ipcc.c
+++ b/drivers/mailbox/stm32-ipcc.c
@@ -513,8 +513,9 @@ static int stm32_ipcc_resume(struct device *dev)
 }
 #endif
 
-static SIMPLE_DEV_PM_OPS(stm32_ipcc_pm_ops,
-			 stm32_ipcc_suspend, stm32_ipcc_resume);
+const struct dev_pm_ops stm32_ipcc_pm_ops = {
+	SET_NOIRQ_SYSTEM_SLEEP_PM_OPS(stm32_ipcc_suspend, stm32_ipcc_resume)
+};
 
 static const struct of_device_id stm32_ipcc_of_match[] = {
 	{ .compatible = "st,stm32mp1-ipcc" },
-- 
2.51.0

